shaders/src/light_indirect.fs

```diff
@@ -575,53 +551,61 @@ void evaluateIBL(const MaterialInputs material, const PixelParams pixel, inout v
     // specular layer
     vec3 Fr = vec3(0.0f);
 
+    SSAOInterpolationCache interpolationCache;
+#if defined(BLEND_MODE_OPAQUE) || defined(BLEND_MODE_MASKED) || defined(MATERIAL_HAS_REFLECTIONS)
+    interpolationCache.uv = uvToRenderTargetUV(getNormalizedViewportCoord().xy);
+#endif
+
     // screen-space reflections
-#if defined(MATERIAL_HAS_REFLECTIONS) && REFLECTION_MODE == REFLECTION_MODE_SCREEN_SPACE
-    vec4 Fssr = vec4(0.0f);
-    // evaluateScreenSpaceReflections will set the value of ssr if there's a hit.
-    // ssr.a contains the reflection's contribution.
-    if (pixel.roughness <= 0.1f && frameUniforms.ssrDistance > 0.0f) {
-        vec3 r = getReflectedVector(pixel, shading_view, shading_normal);
-        Fssr = evaluateScreenSpaceReflections(pixel, r);
+#if defined(MATERIAL_HAS_REFLECTIONS)
+    vec4 ssrFr = vec4(0.0f);
+#if defined(BLEND_MODE_OPAQUE) || defined(BLEND_MODE_MASKED)
+    // do the uniform based test first
+    if (frameUniforms.ssrDistance > 0.0f) {
+        // There is no point doing SSR for very high roughness because we're limited by the fov
+        // of the screen, in addition it doesn't really add much to the final image.
+        // TODO: maybe make this a parameter
+        const float maxPerceptualRoughness = sqrt(0.5);
```

Comment:
Just hardcode 0.125 in case the shader compiler doesn't inline the sqrt()? (and leave a comment about where the value comes from)

Reply:
because it's `const`, which really means `constexpr`, it has no choice but to inline `sqrt`.